ada.html.markdown

```diff
@@ -0,0 +1,367 @@
+with Ada.Text_IO;
+with Stuff;
+
+procedure LearnAdaInY is
+   --  Indentation is 3 spaces.
+
+   --  The most important feature in Ada is the type. Objects have types and an
+   --  object of one type cannot be assigned to an object of another type.
+
+   --  You can, and should, define your own types for the domain you are
+   --  modelling. But you can use the standard types to start with and then
+   --  replace them later with your own types, this could be called a form of
+   --  gradual typing.
+
+   --  The standard types would only really be a good starting point for binding
+   --  to other languages, like C. Ada is the only language with a standardised
+   --  way to bind with [C](https://ada-lang.io/docs/arm/AA-B/AA-B.3),
+   --  [Fortran](https://ada-lang.io/docs/arm/AA-B/AA-B.5/), and even
+   --  [COBOL](https://ada-lang.io/docs/arm/AA-B/AA-B.4/)!
+
+   type Degrees is range 0 .. 360;  --  This is a type. Its underlying logic
+                                    --  is an Integer.
+
+   type Hues is (Red, Green, Blue, Purple, Yellow);  --  So, is this. Here, we
+                                                     --  are declaring an
+                                                     --  Enumeration.
+
+   --  This is a modular type. They behave like Integers that automatically
+   --  wrap around. In this specific case, the range would be 0 .. 359.
+   --  If we added ```+ 1``` to a variable containing the value 359,
+   --  we would receive back 0. They are very useful for arrays
+   type Degrees_Wrap is mod 360;
+
+   --  You can restrict a type's range using a subtype, this makes them
+   --  compatible with each other, i.e. the subtype can be assigned to an
+   --  object of the type, as can be seen below.
+   subtype Primaries is Hues range Red .. Blue;  --  This is a range.
+
+   --  You can define variables or constants like this:
+   --  Var_Name : Type := Value;
+
+   --  10 is a universal integer. These universal numerics can be used with
+   --  any type which matches the base type.
+   Angle : Degrees := 10;
+   Value : Integer := 20;
+   --  New_Angle : Degrees := Value;  --  Incompatible types won't compile.
+   --  New_Value : Integer := Angle;
+
+   Blue_Hue   :          Primaries := Blue;  --  A variable.
+   Red_Hue    : constant Primaries := Red;   --  A constant.
+   Yellow_Hue : constant Hues      := Yellow;
+   Colour_1   : constant Hues      := Red_Hue;
+   --  Colour_2   : constant Primaries := Yellow_Hue;  --  uncomment to compile.
+
+   --  You can force conversions, but then you are warned by the name of the
+   --  package that you may be doing something unsafe.
+   function Degrees_To_Int is new Ada.Unchecked_Conversion
+     (Source => Degrees,   --  Line continuations are indented by 2 spaces.
+      Target => Integer);
+
+   New_Value_2 : Integer := Degrees_To_Int (Angle);  --  Note, space before (.
+
+   --  Ada has a style guide and GNAT will force you to adhere to it, so that
```

Comment:
This is missing a definition of what GNAT is.
Also GNAT doesn't force a style, but it provides options to check coding style.

Reply:
Ok, maybe nag you to fix it.